pkg/utils/mountinfo/mountpoint.go

```diff
@@ -81,14 +82,15 @@ func getBindMounts(mountByPath map[string]*Mount) (bindMountByName map[string][]
 	bindMountByName = make(map[string][]*Mount)
 	for k, m := range mountByPath {
 		var datasetNamespacedName string
+		mCopy := m
```

Comment:
In this change， using  mCopy := m to create a "copy" of a mount point (Mount), where m is a pointer type (*Mount). However, it's important to note that mCopy merely copies the pointer and not the actual data it points to. This means that mCopy and m actually point to the same memory address, and any modifications to mCopy will affect the data pointed to by m. I think we need `deepCopy` to avoid original object modification.

Reply:
In this case, m/mCopy is readonly, the shallow copy `mCopy` intends to address `loop scoping issue` in golang version <= 1.21, ref:  https://go.dev/blog/loopvar-preview